SimpleTotp/TotpProvider.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ReSharper disable StringLiteralTypo
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+[assembly:InternalsVisibleTo("SimpleTotp.Tests")]
+
+namespace SimpleTotp
+{
+    /// <inheritdoc />
+    public class TotpProvider : ITotpProvider
+    {
+        /// <summary>
+        /// Unix Epoch time - in ticks
+        /// </summary>
+        private const long UnixEpochInTicks = 621355968000000000L;
+
+        /// <summary>
+        /// Period for which the same code should be calculated - 30 seconds (in ticks)
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 30 seconds - hence a const
+        /// </remarks>
+        /// </summary>
+        private const long Period = 300000000L;
+
+        /// <summary>
+        /// Number of digits in generated TOTP code
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 6 digits - hence a const
+        /// </remarks>
+        /// </summary>
+        private const int CodeDigits = 6;
```

Comment:
Shouldn't this be a parameter?

Reply:
The reason this is not a parameter is explained literally in the line visible here. While in theory TOTP specification allows the choice of the number of digits, all the major implementations I'm aware of (Google, Microsoft) have it set to strictly 6. If we ever want to support some third-party software that uses a different number of digits, I feel this can be addressed in some future release. KISS for now :)